### PR TITLE
refactor(dart/transform): AnnotationMatcher tests

### PR DESCRIPTION
**Pull request by kegluneq**
/cc @juliemr 

These tests were previously not being run.

Bring them up to modern usage, move them to package:test, and include
them in transform.server.spec.dart.

### CODE REVIEW

**Comment by juliemr** [comment]
the test changes LGTM, but I don't see the change that includes them in transform.server.spec.dart. Did that get left out from the commit?

File: modules_dart/transform/test/transform/transform.server.spec.dart [hunk 1]
```diff
@@ -3,12 +3,14 @@ library angular2.test.transform.transform.server.spec;
...
 import 'package:test/test.dart';
 
+import 'common/annotation_matcher_test.dart' as annotationMatcher;
 import 'common/async_string_writer_tests.dart' as asyncStringWriter;
 import 'common/code/ng_deps_code_tests.dart' as ngDepsCode;
 import 'common/ng_meta_test.dart' as ngMetaTest;
 import 'common/url_resolver_tests.dart' as urlResolver;
 
 main() {
+  group('AnnotationMatcher', annotationMatcher.allTests);
```
Comment (kegluneq) [inline modules_dart/transform/test/transform/transform.server.spec.dart]: @juliemr , this line adds the changes to transform.server.spec.dart

**Comment by juliemr** [comment]
LGTM too.

**Comment by vikerman** [comment]
TAP - 0 affected targets

**Pull request closed without merging**